pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -1052,13 +1052,19 @@ func (ds *Plugin) openDB(cfg *configuration, isReadOnly bool) (*gorm.DB, string,
 	db.SetLogger(gormLogger{
 		log: ds.log.WithField(telemetry.SubsystemName, "gorm"),
 	})
-	db.DB().SetMaxOpenConns(100) // default value
+
+	const maxOpenConns = 100
+	db.DB().SetMaxOpenConns(maxOpenConns)
 	if cfg.MaxOpenConns != nil {
 		db.DB().SetMaxOpenConns(*cfg.MaxOpenConns)
 	}
+	const maxIdleConns = 100
+	db.DB().SetMaxIdleConns(maxIdleConns)
 	if cfg.MaxIdleConns != nil {
 		db.DB().SetMaxIdleConns(*cfg.MaxIdleConns)
 	}
+	const connMaxLifetime = time.Second * 30
+	db.DB().SetConnMaxLifetime(connMaxLifetime)
```

Comment:
The SetConnMaxLifetime function is already called later, based on the ConnMaxLifetime setting. We shouldn't need to make any change here. What we need to adjust is the `ConnMaxIdleTime` property calling `SetConnMaxIdleTime`, maybe with a value of 30s.